### PR TITLE
[DM-27441] Load images list from URL

### PR DESCRIPTION
**Pull request by cbanek**
Allow the retrieval of a list of images from the cachemachine service,
which also tell it which is the recommended tag.

### CODE REVIEW

**Review by rra — APPROVED** [review]
Looks good with some nits.

File: src/nublado2/options.py [hunk 1]
```diff
@@ -51,10 +54,29 @@
...
+
 
 class NubladoOptions(LoggingConfigurable):
-    def show_options_form(self, spawner: Spawner) -> str:
+    async def show_options_form(self, spawner: Spawner) -> str:
         options_config = NubladoConfig().get()["options_form"]
-        images = options_config["images"]
         sizes = options_config["sizes"]
+
+        images_url = options_config.get("images_url", "")
```
[the remark this answers is not in the document]
Comment (cbanek) [reply src/nublado2/options.py:68]: Ah yeah that's where I was having trouble, with the str or none on the typing!  Thanks, will do.

File: src/nublado2/options.py [hunk 2]
```diff
@@ -51,10 +54,29 @@
...
+
         return options_template.render(images=images, sizes=sizes)
+
+    async def _get_images_from_url(self, url: str) -> List[Dict[str, str]]:
+        if not url:
+            return []
+
+        async with session.get(url) as r:
+            if r.status != 200:
+                raise Exception(f"Error {r.status} from {url}")
```
[the remark this answers is not in the document]
Comment (cbanek) [reply src/nublado2/options.py:80]: Yeah, I was thinking about caching it, but then I'd have to deal with caching and invalidation and such, so I decided to not bother at least to start.  I'm wondering what the possible intercluster failure rate would be for transient requests, although so far it doesn't seem too high.  But I agree we can add some caching if there's a problem, although then we have caching. :)

[the remark this answers is not in the document]
Comment (cbanek) [reply src/nublado2/options.py:80]: as for the async with, I think I pulled this out of an example somewhere, but your way does look better, so I'll give that a shot.

File: dev-values.yaml [hunk 3]
```diff
@@ -34,13 +34,8 @@ jupyterhub:
...
+    images_url: "http://cachemachine-dev.cachemachine-dev.svc.cluster.local/cachemachine/jupyter/available"
     images:
-      - name: Recommended
-        image: registry.hub.docker.com/jupyterhub/k8s-singleuser-sample:0.9.0-n358.h11c2aeb8
-      - name: Weekly 44
-        image: registry.hub.docker.com/lsstsqre/sciplat-lab:w_2020_44
-      - name: Weekly 43
-        image: registry.hub.docker.com/lsstsqre/sciplat-lab:w_2020_43
       - name: Weekly 42
         image: registry.hub.docker.com/lsstsqre/sciplat-lab:w_2020_42
```
[the remark this answers is not in the document]
Comment (cbanek) [reply dev-values.yaml:40]: Perfect, thanks indeed!

**Pull request merged**